app/src/main/java/de/westnordost/streetcomplete/quests/internet_access/AddInternetAccess.java

```diff
@@ -0,0 +1,52 @@
+package de.westnordost.streetcomplete.quests.internet_access;
+
+
+import android.os.Bundle;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+
+public class AddInternetAccess extends SimpleOverpassQuestType
+{
+	@Inject public AddInternetAccess(OverpassMapDataDao overpassServer) { super(overpassServer); }
+
+	@Override
+	protected String getTagFilters() {
+		return "nodes, ways, relations with " +
+				" amenity ~ restaurant|cafe|ice_cream|fast_food|bar|pub|biergarten|food_court|cinema|library" +
+				"  or" +
+				" tourism ~ hotel|guest_house|hostel|motel" +
+				" and !internet_access and name";
+	}
+
+	public AbstractQuestAnswerFragment createForm()
+	{
+		return new AddInternetAccessForm();
+	}
+
+	public void applyAnswerTo(Bundle bundle, StringMapChangesBuilder changes)
+	{
+		String osmValue = bundle.getString(AddInternetAccessForm.OSM_VALUE);
+		if (osmValue != null)
+		{
+			changes.add("internet_access", osmValue);
```

Comment:
I made a small mistake while implementing this quest:
I used the value ```wifi``` instead of the correct value ```wlan```
Tagging it as ```internet_access=wifi``` counts as a mistake. I will change this with the next commit.

Reply:
And the wiki says:
> wifi=* ‒ This is an unvoted tag that is in widespread use.